test/token/restricted.sol

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.18;
+pragma abicoder v2;
+
+import {RestrictedTokenFactory, MemberlistFactory} from "src/token/factory.sol";
+import {RestrictedTokenLike} from "src/token/restricted.sol";
+import {MemberlistLike, Memberlist} from "src/token/memberlist.sol";
+import "forge-std/Test.sol";
+
+interface ERC20Like {
+    function balanceOf(address) external view returns (uint256);
+}
+
+contract RestrictedTokenTest is Test {
+    RestrictedTokenLike token;
+    MemberlistLike memberlist;
+
+    function setUp() public {
+        RestrictedTokenFactory tokenFactory = new RestrictedTokenFactory();
+        MemberlistFactory memberlistFactory = new MemberlistFactory();
+
+        token = RestrictedTokenLike(tokenFactory.newRestrictedToken("Some Token", "ST", 18));
+
+        memberlist = MemberlistLike(memberlistFactory.newMemberlist());
+        token.file("memberlist", address(memberlist));
+    }
+
+    function testTransferFromTokensToMemberWorks(uint256 amount, address targetUser, uint256 validUntil) public {
+        vm.assume(validUntil > block.timestamp && targetUser != address(0) && targetUser != address(this));
+
+        memberlist.updateMember(targetUser, validUntil);
+        assertEq(memberlist.members(targetUser), validUntil);
+
+        token.mint(address(this), amount);
```

Comment:
mint is an `auth` function but still allows you to mint to a non-whitelisted address. Is this okay?

Reply:
Also discussed this a bit with Manuel. The only possible reason could be some kind of emergency, but even then if the origin is a ward that can mint, they can also add a member. So I agree and will add it.